ql/src/java/org/apache/hadoop/hive/ql/ddl/table/partition/add/AbstractAddPartitionAnalyzer.java

```diff
@@ -130,4 +131,19 @@ protected void analyzeCommand(TableName tableName, Map<String, String> partition
 
   protected abstract void postProcess(TableName tableName, Table table, AlterTableAddPartitionDesc desc,
       Task<DDLWork> ddlTask) throws SemanticException;
+
+  // Equivalent to acidSinks, but for DDL operations that change data.
+  private DDLDescWithWriteId ddlDescWithWriteId;
+
+  protected void setAcidDdlDesc(DDLDescWithWriteId descWithWriteId) {
+    if (this.ddlDescWithWriteId != null) {
+      throw new IllegalStateException("ddlDescWithWriteId is already set: " + this.ddlDescWithWriteId);
+    }
+    this.ddlDescWithWriteId = descWithWriteId;
+  }
+
+  @Override public DDLDescWithWriteId getAcidDdlDesc() {
```

Comment:
can you fix the formatting? You can import the coding style formatter into your IDE by using dev-support/eclipse-styles.xml to begin with so that such trivial formatting changes don't happen.

Reply:
I have imported the one suggested for Apache Hive development. Need to check why it's not doing its job.